pkg/plugin/processor/registry.go

```diff
@@ -0,0 +1,66 @@
+// Copyright © 2024 Meroxa, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//go:generate mockgen -source=registry.go -destination=mock/processor_creator.go -package=mock -mock_names=processorCreator=ProcessorCreator . processorCreator
+
+package processor
+
+import (
+	"context"
+
+	sdk "github.com/conduitio/conduit-processor-sdk"
+	"github.com/conduitio/conduit/pkg/foundation/cerrors"
+	"github.com/conduitio/conduit/pkg/foundation/log"
+	"github.com/conduitio/conduit/pkg/plugin"
+)
+
+type processorCreator interface {
+	NewProcessor(ctx context.Context, fullName plugin.FullName, id string) (sdk.Processor, error)
+}
+
+type Registry struct {
```

Comment:
There's a bit of a discrepancy now between `plugin.Service` which is the plugin service for connector plugins and this `processor.Registry` which is essentially the same thing for processors.

I'm thinking that we should move that `plugin.Service` into package `pkg/plugin/connector` so it's on the same level as this `Registry` and then either call this `Service` or call that one `Registry`, so it's clear that they are the same thing, one for connector plugins and the other for processor plugins.